osf/migrations/0084_preprint_node_divorce.py

```diff
@@ -0,0 +1,69 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.11.9 on 2018-03-12 18:25
+from __future__ import unicode_literals
+
+from django.db import migrations
+from django.db.models import F
+from itertools import islice
+
+
+def divorce_preprints_from_nodes(apps, schema_editor):
+    Preprint = apps.get_model('osf', 'PreprintService')
+    PreprintContributor = apps.get_model('osf', 'PreprintContributor')
+    # tried to use F() function here but F() doesn't support table joins
+    # instead, using the following to make this transaction atomic
+
+    Preprint.objects.filter(node__isnull=False).select_related(
+        'node', 'node__creator', 'node__title', 'node__description'
+    ).update(title=F('node__title'), description=F('node__description'), creator=F('node__creator'))
+
+    contributors = []
+
+    for preprint in Preprint.objects.filter(node__isnull=False):
+        # use bulk create
+        admin = []
+        write = []
+        read = []
+        for contrib in preprint.node.contributor_set.all():
+            # make a PreprintContributor that points to the pp instead of the node
+            # because there's a throughtable, relations are designated
+            # solely on the through model, and adds on the related models
+            # are not required.
+
+            new_contrib = PreprintContributor(
+                preprint=preprint,
```

Comment:
If you directly import the PreprintService model above, then you should do `preprint_id=preprint.id` and `user_id=contrib.user.id`.